templates/templates/generic-component-busola.yaml

```diff
@@ -0,0 +1,127 @@
+     - ReleaseBranchPattern - pattern for release branch prefix (default: ^release-<supported-releases>-<component-dir-name>$)
+     - runIfChanged - a list of regexps to watch in addition to path (default: [])
+     - resources.memory - memory given to job container (default: 1.5Gi)
+     - resources.cpu - cpu given to job container (default: 0.8)
+  */ -}}
+
+{{ $repositoryName := index (.Values.repository | splitList "/") 2 -}}
+{{- $additionalRunIfChanged := .Values.additionalRunIfChanged | default (list) }}
+{{- $runIfChanged := prepend $additionalRunIfChanged (print "^" .Values.path "/") | join "|" }}
+{{- $componentName := .Values.path | splitList "/" | last }}
+{{- $isNotTestInfra := ne .Values.repository "github.com/kyma-project/test-infra"}}
+{{- $hasReleaseBounds := and (not .Values.since) (not .Values.until) }}
+{{- $releases := matchingReleases .Global.releases .Values.since .Values.until }}
+{{- $isValidForNextRelease := releaseMatches .Global.nextRelease .Values.since .Values.until }}
+{{- $releasesPattern := $releases | join "|" | replace "." "\\." }}
+{{- $releaseBranchReleasesPattern := $hasReleaseBounds | ternary ".*" (print "("  $releasesPattern ")") }}
+{{- $defaultReleaseBranchPattern := (print "^release-" $releaseBranchReleasesPattern "-" $componentName "$") }}
+{{- $defaultReleaseBranchPatternOrEmpty := ne (len $releases) 0 | ternary $defaultReleaseBranchPattern "" }}
+{{- $releaseBranchPattern := .Values.ReleaseBranchPattern | default $defaultReleaseBranchPatternOrEmpty }}
+
+{{- if $isNotTestInfra }}
+test_infra_ref: &test_infra_ref
+  org: kyma-project
+  repo: test-infra
+  path_alias: github.com/kyma-project/test-infra
+{{- end }}
+
+job_template: &job_template
+  decorate: true
+  path_alias: {{ .Values.repository }}
+  max_concurrency: 10
+  labels:
+    preset-dind-enabled: "true"
+    preset-sa-gcr-push: "true"
+    preset-docker-push-repository-{{ .Values.pushRepository }}: "true"
+  run_if_changed: "{{ $runIfChanged }}"
+  spec:
+    containers:
+      - image: eu.gcr.io/kyma-project/test-infra/bootstrap:{{ .Values.bootstrapTag }}
+        securityContext:
+          privileged: true
+        command:
+          - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/build-generic.sh"
+        args:
+          - "/home/prow/go/src/{{ .Values.repository }}/{{ .Values.path }}"
+        resources:
+          requests:
+            memory: {{ .Values.resources.memory | default "1.5Gi" }}
+            cpu: {{ .Values.resources.cpu | default 0.8 }}
+
+presubmits: # runs on PRs
+  {{ .Values.repository | replace "github.com/" "" }}:
+    {{- if $isValidForNextRelease }}
+    - name: pre-{{ $repositoryName }}-{{ .Values.path | replace "/" "-" }}
+      cluster: {{if $.Values.cluster.presubmit}}{{ $.Values.cluster.presubmit }}{{else}}{{fail "Value for cluster not provided"}}{{end}}
+      optional: {{ .Values.optional | default false }}
+      <<: *job_template
```

Comment:
You'd have to move the job template here, or use go's `text/template` functions to move the argument list here. Idk... Maybe you can do something like define new Go template with the job spec instead of anchor and then operate on dynamically specified value. We do similar thing in the [generic.tmpl](https://github.com/kyma-project/test-infra/blob/master/templates/templates/generic.tmpl)
```yaml
{{ define "jobSpec" }}
# prowjob spec
...
spec:
  containers:
    - command: ["/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/build-generic.sh"]
      args:
         - "/home/prow/go/src/{{ .Values.repository }}/{{ .Values.path }}"
         - {{ $ciStage }}
...
{{ end }}
```
And in the actual job do something like:
```
...
{{ $ciStage := "ci-pr" # or ci-master based on the Makefile and pjtype }}
{{ template "jobSpec" }}
...
```